waflog/read.go

```diff
@@ -98,20 +101,33 @@ func (ll *FTWLogLines) CheckLogForMarker(stageID string) []byte {
 	crsHeaderBytes := bytes.ToLower([]byte(config.FTWConfig.LogMarkerHeaderName))
 
 	line := []byte{}
-	// find the last non-empty line
-	for err == nil && len(line) == 0 {
+	lineCounter := 0
+	// Look for the header until EOF or `readLimit` lines at most
+	for err == nil {
+		if lineCounter > readLimit {
+			log.Debug().Msg("aborting search for marker")
+			return nil, true
+		}
+		lineCounter++
 		line, _, err = scanner.LineBytes()
-	}
-	if err != nil {
-		if err == io.EOF {
-			return nil
+		if err != nil {
+			if err == io.EOF {
+				return nil, false
+			}
+			log.Trace().Err(err).Msg("found EOF while looking for log marker")
```

Comment:
This line doesn't seem to be about EOF since that was handled above

Reply:
Thanks. That was some copy pasta.